WebApp/handler/AuthenticationHandler.py

```diff
@@ -1,79 +1,132 @@
+import json
+import tornado
 import requests
+import tornado.web
 
-from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getUsernameForID
-from DIRAC.FrameworkSystem.Client.NotificationClient import NotificationClient
+from tornado.web import HTTPError, RequestHandler
+from tornado.template import Template
 
 from WebAppDIRAC.Lib import Conf
 from WebAppDIRAC.Lib.WebHandler import WebHandler, asyncGen
 
+from DIRAC import S_OK, S_ERROR, gLogger
+from DIRAC.FrameworkSystem.Client.NotificationClient import NotificationClient
+from DIRAC.ConfigurationSystem.Client.Helpers.Resources import getIdPOption
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getUsernameForID
+
+try:
+  from OAuthDIRAC.FrameworkSystem.Client.OAuthManagerClient import OAuthManagerClient
+  oauth = OAuthManagerClient()
+except ImportError:
+  oauth = None
+
 
 class AuthenticationHandler(WebHandler):
 
   AUTH_PROPS = "all"
 
-  # Send mail to administrators
   @asyncGen
   def web_sendRequest(self):
+    """ Send mail to administrators
+    """
     typeAuth = str(self.request.arguments["typeauth"][0])
     loadValue = self.request.arguments["value"]
     addresses = Conf.getCSValue('AdminsEmails')
-    NotificationClient().sendMail(
-        addresses, subject="Request from %s %s" %
-        (loadValue[0], loadValue[1]), body='Type auth: %s, details: %s' %
-        (typeAuth, loadValue))
+    subject = "Request from %s %s" % (loadValue[0], loadValue[1])
+    body = 'Type auth: %s, details: %s' % (typeAuth, loadValue)
+    result = NotificationClient().sendMail(addresses, subject=subject, body=body)
+    self.finish(result)
+
+  @asyncGen
+  def web_getAuthNames(self):
+    """ Get list of enable authentication types
+    """
+    self.finish(Conf.getAuthNames())
 
-  # Get information from web.cfg about auth types
   @asyncGen
   def web_getAuthCFG(self):
+    """ Get option from IdP
```

Comment:
You want to give access to the configuration file without any authorization authentication? Is the CS section contains confident information?

Reply:
It contain IdPs some secret information(like as client_id and client_secret), that not returned for this method